linera-views/src/common.rs

```diff
@@ -505,21 +492,49 @@ pub trait LocalAdminKeyValueStore: Sized {
 }
 
 /// Low-level, asynchronous write and read key-value operations. Useful for storage APIs not based on views.
-pub trait KeyValueStore:
+pub trait RestrictedKeyValueStore:
     ReadableKeyValueStore<Self::Error> + WritableKeyValueStore<Self::Error>
 {
     /// The error type.
     type Error: Debug;
 }
 
 /// Low-level, asynchronous write and read key-value operations, without a `Send` bound. Useful for storage APIs not based on views.
-pub trait LocalKeyValueStore:
+pub trait LocalRestrictedKeyValueStore:
     LocalReadableKeyValueStore<Self::Error> + LocalWritableKeyValueStore<Self::Error>
 {
     /// The error type.
     type Error: Debug;
 }
 
+impl<S: RestrictedKeyValueStore> LocalRestrictedKeyValueStore for S {
+    type Error = <Self as RestrictedKeyValueStore>::Error;
+}
+
+impl<S: KeyValueStore> RestrictedKeyValueStore for S {
+    type Error = <Self as KeyValueStore>::Error;
+}
+
+/// Low-level, asynchronous write and read key-value operations. Useful for storage APIs not based on views.
+pub trait KeyValueStore:
+    ReadableKeyValueStore<Self::Error>
+    + WritableKeyValueStore<Self::Error>
+    + AdminKeyValueStore<Self::Error>
```

Comment:
Do we ever have a writable key-value store that isn't readable? Or an admin one that isn't read-write? Otherwise could we reduce the number of types like this?

```rust
pub trait ReadableKeyValueStore {
    type Error;
    // …
}

pub trait WritableKeyValueStore: ReadableKeyValueStore {
   // Uses the ReadableKeyValueStore implementation's Error type.
   // …
}

pub trait AdminKeyValueStore: WritableKeyValueStore { // Or just KeyValueStore
   // Uses the ReadableKeyValueStore implementation's Error type.
   // …
}
```

Reply:
I would prefer to keep the dependencies as they are. This allows to see more clearly what is needed. I corrected the journaling and LRU code so that the dependencies are clearer.